data_prep.py

```diff
@@ -1,150 +1,330 @@
-import matplotlib.pyplot as plt
+import os
+from datetime import datetime
+from pathlib import Path
+
 import numpy as np
 import pandas as pd
+from loguru import logger
+
+
+class Pipeline:
+    def __init__(
+        self,
+        data_path: str = Path(os.getcwd()) / "WEC2022_Data",
+        data_type: str = "train",
+        model_type: str = "predict_upgrade",
+    ):
+        self.data_path, self.data_type, self.model_type = (
+            data_path,
+            data_type,
+            model_type,
+        )
+        self.df = self.merge_files()
+        self.concat_df_with_oh_encoding()
+
+    def read_csv_file(self, file_prefix) -> pd.DataFrame:
+        return pd.read_csv(
+            self.data_path / f"{file_prefix}_{self.data_type}.csv", sep=";"
+        )
+
+    def read_all_files(self) -> (pd.DataFrame, pd.DataFrame, pd.DataFrame):
+        bkg = (
+            self.read_csv_file("BKG").drop(columns=self.get_bkg_drop_cols())
+            if self.data_type == "train"
+            else self.read_csv_file("BKG")
+        )
+        logger.info(f"{datetime.now()} loaded booking data")
```

Comment:
I dont think we need to specify datetime.now() in f string here. Logger should handle this automaticly